### PR TITLE
poco: add package for the Poco C++ library

### PR DESCRIPTION
**Pull request by KurdyMalloy**
POrtable COmponents is a Modern, powerful open source C++ class libraries
for building network- and internet-based applications that run on desktop,
 server, mobile and embedded systems.

Original Makefile -->
https://dev.openwrt.org/browser/packages/libs/poco/Makefile

add license
add maintainer
update URL to latest github stable version (1.6.2)
change patch to the configure file

Signed-off-by: Jean-Michel Julien <jean-michel.julien@trilliantinc.com>

### CODE REVIEW

File: libs/poco/Makefile [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+PKG_NAME:=poco
+PKG_VERSION:=1.6.2
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/pocoproject/poco.git
+PKG_SOURCE_VERSION:=1cbaf38bbc84159b2ffa87ecc5d8be15d943dc31
```
Comment (yousong) [inline libs/poco/Makefile]: Looks like this commit is on 1.7 branch:  https://github.com/pocoproject/poco/commit/1cbaf3

If it works for you, I would suggest we just use the release tarball from pocoproject.org.  Release versions are supposed to be more stable and maintained.  Making tarballs from git repo requires cloning the whole repo every time.

**Comment by yousong** [comment]
Hi, @KurdyMalloy , I'd like to add that there is really not much benefit in using `Package/$(PKG_INSTALL)`.

 - Almost all other packages just use the name directly
 - The `<name>` in `Package/<name>` does not need to be `PKG_NAME` at all

File: libs/poco/Makefile [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	--no-sharedmemory \
+	--no-wstring \
+	--poquito \
+	--minimal \
+	--shared
+
+MAKE_FLAGS += \
+	CXXFLAGS32="$(TARGET_CXXFLAGS)" \
+	CFLAGS32="$(TARGET_CFLAGS)" \
+	LINKFLAGS32="$(TARGET_LDFLAGS)"
```
Comment (yousong) [inline libs/poco/Makefile]: We also work with 64-bit targets, so I guess this part needs to be reworked.

**Comment by yousong** [comment]
This package does not compile with musl-libc, as only `int strerror_r()` is provided and this will fail the code at https://github.com/pocoproject/poco/blob/develop/Foundation/src/Error.cpp#L73-L74

File: libs/poco/Makefile (hunk 1 above)
Comment (KurdyMalloy) [inline libs/poco/Makefile]: I'm not crazy, at the time I did it, it was 1.6.2, they replaced it with 1.7.0... [pocoproject/poco@18bbff](https://github.com/pocoproject/poco/commit/18bbff)
Anyhow, I will make the change to use the tarball.

**Comment by yousong** [comment]
Hi, @KurdyMalloy , thank you for the work, the patch looks good and it compiles now.  We try to have a commit history where every commit "works", or at least compiles,  so please squash them into a single commit :)

**Comment by KurdyMalloy** [comment]
Done the squash.

Thank you @yousong for taking care of this.

**Pull request merged**